application/models/Survey.php

```diff
@@ -186,7 +186,11 @@ class Survey extends LSActiveRecord implements PermissionInterface
     public function init()
     {
         /** @inheritdoc */
-
+        /* Do not set any default when search, reset gsid */
+        if ($this->scenario == 'search') {
```

Comment:
This part improve a lot https://bugs.limesurvey.org/view.php?id=18556 !
And surely https://bugs.limesurvey.org/view.php?id=18473